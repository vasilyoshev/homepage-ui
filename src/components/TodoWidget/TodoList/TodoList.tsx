import React from 'react';
import { RootState } from 'store';
import { useSelector } from 'react-redux';
import { TodoListItem } from 'components';
import { List } from '@material-ui/core';

export const TodoList: React.FC = () => {
  const todos = useSelector(
    (state: RootState) => state.todos,
  );

  return (
    <List>
      {todos?.map((todo) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
        />
      ))}
    </List>
  );
};
