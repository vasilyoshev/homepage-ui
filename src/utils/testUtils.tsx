import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

// TODO extract mocks to separate files, e.g. widgetsMock.js
const mockState = {
  loading: {},
  widgets: { widgetsProps: [] },
  user: {},
};
const createMockStore = configureStore([thunk]);
const store = createMockStore(mockState);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Providers = ({ children }: any) => (
  <Provider store={store}>
    <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
  </Provider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';

export { customRender as render };
