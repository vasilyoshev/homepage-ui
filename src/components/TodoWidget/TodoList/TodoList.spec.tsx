import { TodoList } from './TodoList';
import { render, screen } from 'utils';

const mockData = [{
  text: 'Walk the dog',
  complete: false,
},
{
  text: 'Make app',
  complete: false,
}];
describe('TodoList', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show list of todos', () => {
    render(<TodoList todos={mockData} toggleComplete={jest.fn()} removeTodo={jest.fn()} />);
    mockData.forEach((todo) => {
      expect(screen.getByText(todo.text)).toBeInTheDocument();
    });
  });
});
