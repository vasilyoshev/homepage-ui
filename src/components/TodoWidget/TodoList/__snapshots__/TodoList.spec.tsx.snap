// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders successfully 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
  />
</DocumentFragment>
`;
